src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -176,7 +176,7 @@ else if (httpCode == 409 && e.getMessage().contains("Operation cannot be fulfill
 
             // We need the pod to be running and connected before returning
             // otherwise this method keeps being called multiple times
```

Comment:
I'm assuming your assessment that my change is incorrect is based on this comment. This threw me at first as well, but if you look at how the `validStates` list is actually used below:

 - it's **not** used for _waiting_ on the pod at all
 - it's used for checking that the pod has entered a "broken" state - if the pod enters a state not on this list, the code _stops_ waiting, the pod is assumed broken, and is destroyed immediately. This is what's causing the bug I'm seeing.

My best guess was that the comment refers to the entire wait loop, not just the `validStates` list that it's immediately above.